### PR TITLE
First review.

### PR DESCRIPTION
**Pull request by SigmaSam**
We recreated the bubble sorting algorithm with Ruby and adds as required, a version that uses the yield method to summon a block of code in the function as an argument.

live version
https://repl.it/join/nobpzbtz-marcoshdezcam

### CODE REVIEW

**Review by mikenath223 — CHANGES_REQUESTED** [review]
### Hi Team 👋

You've both obviously worked hard on this. 👍

- [x] - However you are expected to build the bubble_sort method, kindly include the bubble_sort method.

- [x] In addition, the bubble_sort_by method doesn't return the right result, kindly fix this and test to ensure it returns the right result.

Kindly fix these issues and request another review👍
Feel free to contact me on slack(michgolden) if you need further clarification.

HAPPY CODING👨‍💻

**Comment by marcoshdezcam** [comment]
> ### Hi Team wave
> 
> You've both obviously worked hard on this. +1
> 
>     * [ ]  - However you are expected to build the bubble_sort method, kindly include the bubble_sort method.
> 
>     * [ ]  In addition, the bubble_sort_by method doesn't return the right result, kindly fix this and test to ensure it returns the right result.
> 
> 
> Kindly fix these issues and request another review+1
> Feel free to contact me on slack(michgolden) if you need further clarification.
> 
> HAPPY CODINGman_technologist

Issues solved! Thank you!

**Review by mikenath223 — CHANGES_REQUESTED** [review]
### Hi again Team 👋

Kindly check [this link](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/57421588/) to see the project requirements please take note of the section Microverse Adjustments.
You are expected to build two(2) methods. The bubble_sort and the bubble_sort_by method.

I apologize if I wasn't clear in the first review.

- [ ] Also the nature of the bubble_sort method is that it sorts an array in the default ascending order if no block is passed to it, but in yours, if no block is passed it throws an error.

Kindly fix these issues and request another review👍
Feel free to contact me on slack(mike) if you need further clarification.

HAPPY CODING👨‍💻

**Comment by marcoshdezcam** [comment]
> ### Hi again Team wave
> 
> Kindly check [this link](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/57421588/) to see the project requirements please take note of the section Microverse Adjustments.
> You are expected to build two(2) methods. The bubble_sort and the bubble_sort_by method.
> 
> I apologize if I wasn't clear in the first review.
> 
>     * [ ]  Also the nature of the bubble_sort method is that it sorts an array in the default ascending order if no block is passed to it, but in yours, if no block is passed it throws an error.
> 
> 
> Kindly fix these issues and request another review+1
> Feel free to contact me on slack(mike) if you need further clarification.
> 
> HAPPY CODINGman_technologist

Solved!

File: bubble_sort.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    while swap == true
+      swap = false
+      (numbers.length - 1).times do |i|
+        if numbers[i] > numbers[i + 1]
+          numbers[i], numbers[i + 1] = numbers[i + 1], numbers[i]
+          swap = true
+        end
+      end
+    end
+    print numbers
```
Comment (mosesogwo) [inline bubble_sort.rb]: Return numbers, do not print

File: bubble_sort.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  swap = true
+  if array.is_a?(Array) && array.any?
+    while swap == true
+      swap = false
+      (array.size - 1).times do |i|
+        comparator = yield array[i], array[i + 1]
+        if comparator
+          array[i], array[i + 1] = array[i + 1], array[i]
+          swap = true
+        end
```
Comment (mosesogwo) [inline bubble_sort.rb]: You should check if the comparator is greater than zero, before proceeding to swap.
What you have currently will proceed to swap every time except when comparator value is false or nil.

**Review by mosesogwo — CHANGES_REQUESTED** [review]
**Hello! :wave:, you're doing well. Just a little fix and you'll be fit to :rocket:**
- You should return sorted array values and not print them.
- Your bubble_sort_by doesn't return the correct result yet. See my comments on your code.


_**Kindly make those changes and submit another code review request.**_

**Any questions? You can reach me on Slack (Moses Ogwo)**

**Cheers:stars:**

### Status: Changes Requested.

**Comment by Dandush03** [comment]
# Status: Project Approved 

## Good luck in your next assignment

**Pull request merged**